# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsContentAreaDragDrop.cpp
# Commit: 32eeee79d628
# Full Hash: 32eeee79d628dade8109454c636d5f8aaff0c93e
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 04:41:59
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 7 - Use getter methods to access mBrowserParent and mBrowserBridgeChild. r=farre
#   
#   This prepares nsFrameLoader for replacing mBrowserParent and mBrowserBridgeChild
#   with a common interface by making special case code use a getter method instead
#   of direct access.
# ==============================================================================

diff -r 07678a2fa9e7 -r 32eeee79d628 dom/base/nsContentAreaDragDrop.cpp
--- a/dom/base/nsContentAreaDragDrop.cpp	Wed Apr 24 23:52:47 2019 -0500
+++ b/dom/base/nsContentAreaDragDrop.cpp	Thu Apr 25 12:36:16 2019 -0500
@@ -552,11 +552,11 @@
     if (flo) {
       RefPtr<nsFrameLoader> fl = flo->GetFrameLoader();
       if (fl) {
-        BrowserParent* tp = static_cast<BrowserParent*>(fl->GetRemoteBrowser());
-        if (tp) {
+        BrowserParent* bp = fl->GetBrowserParent();
+        if (bp) {
           // We have a BrowserParent, so it may have data for dnd in case the
           // child process started a dnd session.
-          tp->AddInitialDnDDataTo(aDataTransfer, aPrincipal);
+          bp->AddInitialDnDDataTo(aDataTransfer, aPrincipal);
         }
       }
     }